src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -329,15 +317,27 @@ private InitValueView getInitValue(MethodTransformerContext context, InitCodeNod
         initValue.initialValue(
             context.getTypeTable().getZeroValueAndSaveNicknameFor(item.getType()));
       }
-
       return initValue.build();
     }
   }
 
   private static List<String> getFormatFunctionArgs(
-      List<String> varList, InitValueConfig initValueConfig) {
+      MethodTransformerContext context, Field field, InitValueConfig initValueConfig) {
     List<String> formatFunctionArgs = new ArrayList<>();
-    for (String entityName : varList) {
+
+    Set<String> entityNames;
+    boolean shouldUseCollectionConfigEntityNames =
+        !context.getFeatureConfig().useResourceNameFormatOption(field)
+            || initValueConfig.hasFormattingConfigInitialValues();
+    if (shouldUseCollectionConfigEntityNames) {
```

Comment:
As discussed offline, I will change the proto and revert the fix in the java code.